cpp/src/replace/replace.cu

```diff
@@ -87,6 +87,169 @@ __device__ auto get_new_value(cudf::size_type idx,
   return thrust::make_pair(new_value, output_is_valid);
 }
 
+__device__ int get_new_string_value(cudf::size_type idx,
+                                     cudf::column_device_view& input,
+                                     cudf::column_device_view& values_to_replace,
+                                     cudf::column_device_view& replacement_values) {
+  cudf::string_view inputString = input.element<cudf::string_view>(idx);
+  int match = -1;
+  for (int i = 0; i < values_to_replace.size(); i++) {
+    cudf::string_view valueString = values_to_replace.element<cudf::string_view>(i);
+    if (inputString == valueString)
+      match = i;
+  }
+  return match;
+}
+
+template<bool input_has_nulls, bool replacement_has_nulls>
+__global__ void replace_strings_first_pass(cudf::column_device_view input,
+                                           cudf::column_device_view values_to_replace,
+                                           cudf::column_device_view replacement,
+                                           cudf::mutable_column_device_view offsets,
+                                           cudf::mutable_column_device_view indices) {
+  cudf::size_type nrows = input.size();
+  cudf::size_type i = blockIdx.x * blockDim.x + threadIdx.x;
+  while (i < nrows) {
```

Comment:
Factor out repetitive code to simplify this kernel.

Reply:
Done.